File: packages/go/luna/auth/api_key.go
package auth

import (
	"fmt"
	"regexp"
)

// APIKeyAuth implements API key authentication
type APIKeyAuth struct {
	apiKey string
}

var apiKeyPattern = regexp.MustCompile(`^lk_(live|test|dev)_[a-zA-Z0-9]{32}$`)

// NewAPIKeyAuth creates a new API key authentication provider
func NewAPIKeyAuth(apiKey string) (*APIKeyAuth, error) {
	if apiKey == "" {
		return nil, fmt.Errorf("auth: API key is required")
	}
	if !apiKeyPattern.MatchString(apiKey) {
		return nil, fmt.Errorf("auth: invalid API key format, expected: lk_<env>_<key>")
	}
	return &APIKeyAuth{apiKey: apiKey}, nil
}

// GetHeaders returns authorization headers with the API key
func (a *APIKeyAuth) GetHeaders() (map[string]string, error) {
	return map[string]string{
		"Authorization": "Bearer " + a.apiKey,
	}, nil
}

// NeedsRefresh returns false as API keys don't expire
func (a *APIKeyAuth) NeedsRefresh() bool {
	return false
}

// Refresh is a no-op for API keys
func (a *APIKeyAuth) Refresh() error {
	return nil
}

var _ Provider = (*APIKeyAuth)(nil)

File: packages/typescript/src/auth/api-key.ts
import { AuthenticationError } from '../errors/base.js';
import { ErrorCode } from '../errors/codes.js';
import type { AuthProvider } from './types.js';

/**
 * API Key authentication provider
 *
 * @example
 * ```typescript
 * const auth = new ApiKeyAuth('lk_live_xxxx');
 * const headers = await auth.getHeaders();
 * // { 'X-Luna-Api-Key': 'lk_live_xxxx' }
 * ```
 */
export class ApiKeyAuth implements AuthProvider {
    private readonly apiKey: string;

    constructor(apiKey: string) {
        if (!apiKey) {
            throw new AuthenticationError({
                code: ErrorCode.AUTH_INVALID_KEY,
                message: 'API key is required',
                requestId: 'local',
            });
        }
        if (!this.isValidApiKey(apiKey)) {
            throw new AuthenticationError({
                code: ErrorCode.AUTH_INVALID_KEY,
                message: 'Invalid API key format. Expected: lk_<env>_<key>',
                requestId: 'local',
            });
        }
        this.apiKey = apiKey;
    }

    async getHeaders(): Promise<Record<string, string>> {
        return {
            'Authorization': `Bearer ${this.apiKey}`,
        };
    }

    needsRefresh(): boolean {
        return false; // API keys don't expire
    }

    /**
     * Validate API key format
     * Format: lk_<environment>_<32 chars>
     */
    private isValidApiKey(key: string): boolean {
        const pattern = /^lk_(live|test|dev)_[a-zA-Z0-9]{32}$/;
        return pattern.test(key);
    }
}
